libgraphics/include/pivot/graphics/VulkanAllocator.hxx

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include <vk_mem_alloc.hpp>
+#include <vulkan/vulkan.hpp>
+
+#include "pivot/graphics/types/AllocatedBuffer.hxx"
+#include "pivot/graphics/types/AllocatedImage.hxx"
+#include "pivot/graphics/types/common.hxx"
+#include "pivot/graphics/vk_utils.hxx"
+
+namespace pivot::graphics
+{
+
+/// @class VulkanAllocator
+/// @brief Wrapper around VMA
+///
+/// This class store the vma::Allocator object and expose it with simplified function to use it. It also allow us to
+/// monitor its usage
+class VulkanAllocator
+{
+public:
+    /// Hold the memory usage of the allocator
+    struct GPUMemoryStats {
+        /// The amount of used bytes
+        uint64_t used = 0;
+        /// the amount of available bytes
+        uint64_t free = 0;
+    };
+
+public:
+    /// Constructor
+    VulkanAllocator();
+    /// Destructor
+    ~VulkanAllocator();
+
+    /// Create the vma::Allocator object
+    void init(const vma::AllocatorCreateInfo &info);
+    /// Destory the vma::Allocator object. Does not free any allocated memory
+    void destroy() { allocator.destroy(); }
+
+    /// Create a buffer.
+    AllocatedBuffer createBuffer(const vk::DeviceSize &allocSize, const vk::BufferUsageFlags &usage,
+                                 const vma::MemoryUsage &memoryUsage, const vma::AllocationCreateFlags &flags = {});
```

Comment:
Try to give POD by value instead of giving them by reference.

Reply:
What do you mean by that ?